CAIPs/caip-x.md

```diff
@@ -0,0 +1,339 @@
+  },
+  "eip155:137": {
+    "methods": ["eth_sendTransaction"],
+    "notifications": ["chainChanged"],
+    "accounts": ["eip155:137:0xdef456"]
+  }
+}
+```
+
+**Example Request:**
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "provider_revoke",
+  "params": {
+    "scopes": {
+      "eip155:1": {
+        "methods": ["eth_sendTransaction"],
+        "notifications": ["chainChanged"],
+        "accounts": ["eip155:1:0xabc123"]
+      }
+    }
+  }
+}
+```
+
+**Example Response:**
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "success": true,
+    "updatedScopes": {
+      "eip155:1": {
+        "methods": ["eth_signTransaction"],
+        "notifications": ["accountsChanged"],
+        "accounts": ["eip155:1:0xdef456"]
+      },
+      "eip155:137": {
+        "methods": ["eth_sendTransaction"],
+        "notifications": ["chainChanged"],
+        "accounts": ["eip155:137:0xdef456"]
+      }
+    }
+  }
+}
+```
+
+**Explanation:**
+
+- The `provider_revoke` method removes the specified methods, notifications, and accounts from the current session's permissions. If the scope no longer has any methods, notifications, or accounts after revocation, it is removed from the session. If no scopes remain in the session, the session is considered closed. If no accounts remain in a scope, and only write methods the entire scope is removed.
+
+#### `provider_getSession` // open to a different name here!
+
+Retrieves the current permissions of an existing session.
+
+**Parameters:**
+
+- `sessionId` (string, optional): The session identifier.
+
+**Initial Session Scopes:**
+
+```json
+{
+  "eip155:1": {
+    "methods": ["eth_signTransaction"],
+    "notifications": ["accountsChanged"],
+    "accounts": ["eip155:1:0xabc123"]
+  },
+  "eip155:137": {
+    "methods": ["eth_sendTransaction"],
+    "notifications": ["chainChanged"],
+    "accounts": ["eip155:137:0xdef456"]
+  }
+}
+```
+
+**Example Request:**
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "provider_getSession",
+  "params": {}
+}
+```
+
+**Example Response:**
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "scopes": {
+      "eip155:1": {
+        "methods": ["eth_signTransaction"],
+        "notifications": ["accountsChanged"],
+        "accounts": ["eip155:1:0xabc123"]
+      },
+      "eip155:137": {
+        "methods": ["eth_sendTransaction"],
+        "notifications": ["chainChanged"],
+        "accounts": ["eip155:137:0xdef456"]
+      }
+    }
+  }
+}
+```
+
+**Explanation:**
+
+- The `provider_getSession` method returns the current permissions for the session. It lists all scopes along with their methods, notifications, and accounts.
+
+### Events
+
+#### `providerAuthorizationChanged`
+
+This event is published by the wallet to notify the caller/dapp of updates to the session authorization scopes. The event payload indicates how the scopes have changed, showing additions and removals in the permissions.
```

Comment:
Here I would actually prefer the whole config be sent rather than a delta/patch, for clarity and security.  Maybe a privacy wizard could convince me that's not a worthwhile tradeoff, but if nothing else, it makes feature parity with status quo easier to prove, since whole-configs get passed back and forth currently.